library/network/src/modules/firewalld_wrapper.rb

```diff
@@ -101,11 +102,86 @@ def remove_port(port_or_range, protocol, interface)
       zone.remove_port(port)
     end
 
+    # Check whether the firewalld service is enable or not
+    #
+    # @return [Boolean] true if it is enable; false otherwise
+    def is_enabled
+      firewalld.enabled?
+    end
+
+    # Return true if the logging config or any of the zones where modified
+    # since read
+    #
+    # @return [Boolean] true if the config was modified; false otherwise
+    def is_modified
+      firewalld.modified?
+    end
+
+    # Evaluate the zone name of an interface
+    #
+    # @param interface [String] interface name
+    #
+    # @return [String] zone name (nil; not found)
+    def zone_name_of_interface(interface)
+      zone = interface_zone(interface)
+      return nil unless zone
+      zone.name
+    end
+
+    # Check if the service belongs to the zone
+    #
+    # @param service [String] service name
+    # @param zone [String] zone name
+    #
+    # @return [Boolean] true if service is in zone
+    def is_service_in_zone(service, zone_name)
+      zone = firewalld.find_zone(zone_name)
+      return false unless zone
+      zone.services.include?(service)
+    end
+
+    # Return an array with all the known (sysconfig configured) firewalld
+    # interfaces.
+    #
+    # @return [Array<Hash>] known interfaces
+    #        e.g. [{ "id":"eth0", "name":"Askey 815C", "zone":"EXT"} , ... ]
+    def all_known_interfaces
+      ret = []
+      Y2Firewall::Firewalld::Interface.known.each do |interface|
+        ret << { "id" => interface.name, "zone" => interface.zone.name,
+                "name" => interface.device_name }
+      end
+      ret
+    end
+
+    # sets status for several services on several network interfaces.
+    #
+    # @param	list <string> service ids
+    # @param	list <string> network interfaces
+    # @param	boolean new status of services
+    def set_services(services, interfaces, status)
```

Comment:
mmm not sure about this method name, I would maybe suggest something like `modify_interface_services`